src/utils/proptypes.ts

```diff
@@ -0,0 +1,25 @@
+import { ComponentType, ElementType } from 'react'
+
+const isKeyOf = <T>(o: keyof T) => !!o
```

Comment:
does not check anything on runtime ?!

Reply:
After some search, I changed my solution to something more permissive but working better at runtime